lib/devise/competitions_failure_app.rb

```diff
@@ -0,0 +1,31 @@
+module Devise
+  class CompetitionsFailureApp < Devise::FailureApp
+    protected
+
+    def i18n_message(default = nil)
+      domain          = params[:registered_user][:uid].split("@").last
```

Comment:
`user_domain` or `user_account_domain` might be clearer. if you copied this from my old code, sorry.

Reply:
also, maybe this file should be called `saml_login_failure.rb` rather than `competitions_failure.rb`